File: server/Models/Moit.js
const mongoose = require('mongoose');

const submenuSchema = mongoose.Schema({
    nums: { type: Number, required: true },
    title: { type: String, required: true },
    make_by: { type: String },
    path: { type: String },
    pdfurl: { type: String },
    fc_year: { type: Number }, 
    count_view: { type: Number }, 
    count_download: { type: Number }, 
}, { timestamps: true });

const childrenSchema = mongoose.Schema({
    nums: { type: Number, required: true },
    title: { type: String, required: true },
    make_by: { type: String },
    path: { type: String },
    pdfurl: { type: String },
    check: { type: String }, 
    fc_year: { type: Number }, 
    count_view: { type: Number }, 
    count_download: { type: Number },
    subtitle: { type: [submenuSchema], default: [] } 
}, { timestamps: true });

const menuSchema = mongoose.Schema({
    nums: { type: Number, required: true },
    fc_year: { type: Number, required: true },
    title: { type: String, required: true },
    childrens: { type: [childrenSchema], default: [] } 
}, { timestamps: true });

module.exports = mongoose.model('moit', menuSchema);

File: Go_api/models/moit.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Subtitle struct {
	ID primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
	Num float64 `bson:"nums" json:"nums"`
	Title string `bson:"title" json:"title"`
	Make_by string `bson:"make_by" json:"make_by"`
	Path string `bson:"path" json:"path"`
    Pdfurl string `bson:"pdfurl" json:"pdfurl"`
	Fc_year string `bson:"fc_year" json:"fc_year"`
    Count_view int32 `bson:"count_view" json:"count_view"`
    Count_download int32 `bson:"count_download" json:"count_download"`
}

type Childrens struct {
	ID primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
	Nums int32 `bson:"nums" json:"nums"`
	Title string `bson:"title" json:"title"`
	Make_by string `bson:"make_by" json:"make_by"`
    Path string `bson:"path" json:"path"`
    Pdfurl string `bson:"pdfurl" json:"pdfurl"`
	Check string `bson:"check" json:"check"`
    Fc_year string `bson:"fc_year" json:"fc_year"`
    Count_view int32 `bson:"count_view" json:"count_view"`
    Count_download int32 `bson:"count_download" json:"count_download"`
	Subtitle []Subtitle `bson:"subtitle" json:"subtitle"`
}

type Moit struct {
	ID primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
	Nums int32 `bson:"nums" json:"nums"`
	Title string `bson:"title" json:"title"`
	Fc_year int32 `bson:"fc_year" json:"fc_year"`
	Childrens []Childrens `bson:"childrens" json:"childrens"`
}